x/nft-transfer/types/trace.go

```diff
@@ -0,0 +1,180 @@
+package types
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"sort"
+	"strings"
+
+	tmbytes "github.com/tendermint/tendermint/libs/bytes"
+	tmtypes "github.com/tendermint/tendermint/types"
+
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	host "github.com/cosmos/ibc-go/v5/modules/core/24-host"
+)
+
+// ParseHexHash parses a hex hash in string format to bytes and validates its correctness.
+func ParseHexHash(hexHash string) (tmbytes.HexBytes, error) {
+	hash, err := hex.DecodeString(hexHash)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := tmtypes.ValidateHash(hash); err != nil {
+		return nil, err
+	}
+
+	return hash, nil
+}
+
+// GetClassPrefix returns the receiving class prefix
+func GetClassPrefix(portID, channelID string) string {
+	return fmt.Sprintf("%s/%s/", portID, channelID)
+}
+
+// RemoveClassPrefix returns the unprefixed classID.
+// After the receiving chain receives the packet,if isAwayFromOrigin=false, it means that nft is moving
+// in the direction of the original chain, and the portID/channelID prefix of the sending chain
+// in trace.path needs to be removed
+func RemoveClassPrefix(portID, channelID, classID string) string {
+	classPrefix := GetClassPrefix(portID, channelID)
+	return classID[len(classPrefix):]
+}
+
+// IsAwayFromOrigin determine if non-fungible token is moving away from
+// the origin chain (the chain issued by the native nft).
+// Note that fullClassPath refers to the full path of the unencoded classID.
+// The longer the fullClassPath, the farther it is from the origin chain
+func IsAwayFromOrigin(sourcePort, sourceChannel, fullClassPath string) bool {
+	prefixClassID := GetClassPrefix(sourcePort, sourceChannel)
+	if !strings.HasPrefix(fullClassPath, prefixClassID) {
+		return true
+	}
+	return fullClassPath[:len(prefixClassID)] != prefixClassID
```

Comment:
this is equal to `return !strings.HasPrefix(fullClassPath, prefixClassID)`

Reply:
it looks like it: https://cs.opensource.google/go/go/+/master:src/strings/strings.go;l=451 ?
@devashishdxt can it be simplified or is there a reason for checking like this?

Reply:
I'll simplify

Reply:
Done